test/tests/functional/pbs_qstat.py

```diff
@@ -47,31 +47,34 @@ class QstatTest(TestFunctional):
     """
 
     def setUp(self):
-        if len(self.servers) != 2 or self.server.client in self.servers:
-            self.skipTest("This test needs two servers and one client")
+        if self._testMethodName == "test_qstat_req_server":
+            if len(self.servers) != 2 or self.server.client in self.servers:
+                self.skipTest("This test needs two servers and one client")
+            # Because of a bug in PTL, having moms on respective server hosts
+            # don't work, so the server hosts need to be passed as nomom hosts
+            svrnames = self.servers.keys()
+            if "nomom" not in self.conf or \
+                    svrnames[0] not in self.conf["nomom"] or \
+                    svrnames[1] not in self.conf["nomom"]:
+                self.skipTest("This test needs the server hosts to be passed"
+                              " as nomom hosts: -p servers=<host1>:<host2>,"
+                              "nomom=<host1>:<host2>")
         TestFunctional.setUp(self)
+
+    def test_qstat_req_server(self):
         self.server1 = self.servers.values()[0]
         self.server2 = self.servers.values()[1]
         a = {'scheduling': 'false', 'flatuid': 'true'}
         self.server1.manager(MGR_CMD_SET, SERVER, a)
         self.server2.manager(MGR_CMD_SET, SERVER, a)
 
-    def test_qstat_req_server(self):
         j = Job(TEST_USER)
         jid = self.server1.submit(j)
         destination = '@%s' % self.server2.hostname
         self.server1.movejob(jobid=jid, destination=destination)
-        self.client_conf = self.du.parse_pbs_config(
-            hostname=self.server.client)
-        self.du.set_pbs_config(self.server.client, confs={'PBS_SERVER': ''})
-        self.server1.status(JOB, id=jid+'@%s' % self.server2.hostname,
+        self.server1.status(JOB, id=jid + '@%s' % self.server2.hostname,
```

Comment:
hmm .. I kind of feel like PTL should take care of this, not the test writer. API mode shouldn't require test writers to write their tests any differently right?

Reply:
@agrawalravi90 and @bhroam I believe the API mode is broken or not widely used today in PTL. Also we might need a separate discussion about making PTL generic for API vs CLI mode. For now I suggest follow what other tests might have been doing, i.e., skipping the tests in API mode if not supported, since today PTL methods do not distinguish by itself.

Reply:
@anamikau thanks for your advice, I'm pretty sure that API mode is broken and has not been supported for a while now. Can you please point me to a test that skips for API mode? I'm not sure how to check for it. Thanks.

Reply:
@agrawalravi90 you can add something like

        _m = self.server.get_op_mode()
        if _m != PTL_CLI:
            self.skipTest("Test only supported for CLI mode")